corfudb-tools/src/main/java/org/corfudb/browser/CorfuStoreBrowser.java

```diff
@@ -266,6 +275,98 @@ public int dropTable(String namespace, String tablename) {
         return tableSize;
     }
 
+    public int editRecord(String namespace, String tableName,
+        String keyToEdit, String newRecord) {
+        verifyNamespaceAndTablename(namespace, tableName);
+
+        if (keyToEdit == null) {
+            throw new IllegalArgumentException("Please specify the key to " +
+                "edit --keyToEdit");
+        }
+
+        if (newRecord == null) {
+            throw new IllegalArgumentException("Please specify the newRecord " +
+                "--newRecord");
+        }
+
+        log.info("\n======================\n");
+        String fullName = TableRegistry.getFullyQualifiedTableName(namespace,
+            tableName);
+        UUID streamUUID = UUID.nameUUIDFromBytes(fullName.getBytes());
+        CorfuTable<CorfuDynamicKey, CorfuDynamicRecord> table = getTable(
+            namespace, tableName);
+
+        TableName tableNameProto = TableName.newBuilder().setTableName(tableName)
+            .setNamespace(namespace).build();
+
+        CorfuStoreMetadata.KeyValueDefaultRecord keyValueDefaultRecord =
+            runtime.getTableRegistry().getTableNameToKeyValueTypeTable().get(tableNameProto);
+        Any defaultKeyAny = keyValueDefaultRecord.getKey();
+        Any defaultValueAny = keyValueDefaultRecord.getValue();
+
+        DynamicMessage keyMsg = createMsgFromJson(defaultKeyAny, keyToEdit);
+        DynamicMessage newValueMsg = createMsgFromJson(defaultValueAny,
+            newRecord);
+
+        if (keyMsg == null || newValueMsg == null) {
+            return 0;
+        }
+
+        for (CorfuDynamicKey key : table.keySet()) {
+            if (Objects.equals(keyMsg.toString(), key.getKey().toString())) {
+                log.info("Editing record with Key {} in table {} and " +
+                        "namespace {} with new record {}.  Stream Id {}.",
+                    keyToEdit, tableName, namespace, newRecord, streamUUID);
+                String payloadTypeUrl = table.get(key).getPayloadTypeUrl();
+                String metadataTypeUrl = table.get(key).getMetadataTypeUrl();
+                DynamicMessage metadata = table.get(key).getMetadata();
+                table.put(key, new CorfuDynamicRecord(payloadTypeUrl,
+                    newValueMsg, metadataTypeUrl, metadata));
+                log.info("\n======================\n");
```

Comment:
This write should be transactional otherwise it can cause silent failures. For example, imagine a vertical trying to edit a key while this command is executed. Both operations won't fail, but one of them should have.

Reply:
done